### PR TITLE
Amend Nagios/Ansible Config Role To Create Service Groups Config File 

### LINKED ISSUE #2865: Nagios : Amend Nagios/Ansible Config Role To Create Service Groups Config File
Part of epic : [#2445](https://github.com/adoptium/infrastructure/issues/2445) ( Task 6.3 )

Update the Nagios\_Config playbook to create the servicegroups.cfg file

The script should parse the ansible inventory ( similar to issue [#2629](https://github.com/adoptium/infrastructure/issues/2629) )

And create a config file which contains a service group entry for each of the group types in the inventory, ie the first stanza for each of the build, test & dockerhost level 1 stanzas.

The format of the servicegroups.cfg file should be as below.. ( please keep the default servicegroup Web\_Sites as it is required for Nagios )

define servicegroup{  
       servicegroup\_name       Web\_Sites  
       alias                   Web Sites  
       }

define servicegroup{  
       servicegroup\_name       Docker\_Hosts  
       alias                   Docker Hosts  
       }

### PR DESCRIPTION
**Pull request by lumuchris256**
Updated the Nagios_Config playbook to create the servicegroups.cfg file that contains a service group entry for each of the group types in the inventory

fixes #2865 
signed off by @lumuchris256 
cc @steelhead31 @Haroon-Khel @smlambert

### CODE REVIEW

File: ansible/playbooks/nagios/roles/Nagios_Config/scripts/Create_Nagios_Servicegroups_config_file.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+                        export[host_type]['hosts'].append(hostname)
+                        hostvars = {}
+
+                        service_list=Nagios_Service_Types.split(" ")
+
+                for service in service_list:
+                       if host_type==service:
+                           formatted_name = host_type+'-'+provider_name+'-'+host
+                           servicegroup_name = host_type + '-' + provider_name
+                           servicegroup_alias = host_type + '-' + provider_name + '-' + host
```
Comment (steelhead31) [inline ansible/playbooks/nagios/roles/Nagios_Config/scripts/Create_Nagios_Servicegroups_config_file.py:87]: I think these lines need looking at for the failure.

**Review by Haroon-Khel — APPROVED** [review]
The changes youve made look good, the python script now runs for me to produce

```
define servicegroup{
  servicegroup_name Web_Servers
  alias Web_Servers
}
define servicegroup{
  servicegroup_name build_Servers
  alias build_Servers
}
define servicegroup{
  servicegroup_name test_Servers
  alias test_Servers
}
define servicegroup{
  servicegroup_name dockerhost_Servers
  alias dockerhost_Servers
}
```
When `build test dockerhost` are passed in as service groups

Unless @steelhead31 has any further objections, this looks good to me

**Review by steelhead31 — APPROVED** [review]
Re-added my relative path changes, but the script works as expected... note that the servicegroups config file now ends up in /usr/local/nagios/etc/objects  , whereby the individual server configs end up in /usr/local/nagios/etc/servers

**Pull request merged**